### ISSUE TITLE
[docs]: Add notes regarding provisioner management when interacting with step-ca in a docker container

### ISSUE THREAD

**Issue by crumpetcrusher** [smallstep/cli#693]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to document this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Affected area/feature

Following through with the docker [Getting Started](https://hub.docker.com/r/smallstep/step-ca) and [ACME Basics](https://smallstep.com/docs/step-ca/acme-basics) pages, when running step-ca in a docker container and running step-cli outside of the container, you will encounter the following error when attempting to add the provisioner:
```
step ca provisioner add acme --type ACME
No admin credentials found. You must login to execute admin commands.
✔ Please enter admin name/subject (e.g., name@example.com): admin
✔ Provisioner: admin (JWK) [kid: tSqHR-XrFV6k1Zr7Ejv0AMNC2YluOwb99aAlRfuYoZI]
Please enter the password to decrypt the provisioner key: ...
json: cannot unmarshal number into Go value of type ca.AdminClientError
```

This has been cleared up in a discord message from @dopey but would be beneficial to have in the sections referencing `add provisioner` commands that it must be ran where the step-ca server is running (as noted on the [provisioners](https://smallstep.com/docs/step-ca/provisioners/#provisioner-management) page), i.e., using `docker exec` instead.

An issue regarding improved console error messages from the CLI relates to this as well, linked in the discord recently: https://github.com/smallstep/cli/issues/684 

https://discord.com/channels/837031272227930163/841249977699401759/982701928989458492

> ...this indicates that the cli is trying to use the Remote Configuration Management API to perform the `provisioner add` action.

> By default your CA will be configured not to allow remote configuration management - which means the provisioners will be stored in the `ca.json` file and the only way to update the provisioners is to run commands locally (wherever `step-ca` is being run).

> So, assuming you have not intentionally enabled Remote Confguration [sic] Management, the cli is attempting to use remote provisioner management because it does not know where to find your `ca.json` file. You must run `step ca provisioner add ...` from the same server where `step-ca` is running, so that the `ca.json` can be edited. The default location for the `ca.json` is `$(step path)/config/ca.json`. You can also use `--ca-config` to point directly to the `ca.json` file, if it is not in the default location. If the `ca.json` file cannot be found (either in the default location or as directed by `--ca-config`) then the CLI will attempt to use remote config management. Which is why you see the `ca.AdminClientError` popping up. (remote configuration management uses the `AdminClient`).

**Comment by dopey** [comment]
I'm hopeful that by closing #684 we can address this in a way that won't need additional documentation. In the mean time, where do you think it would be useful to add these notes @crumpetcrusher? Is there a place you would have been likely to find it? linking @tashian

**Comment by crumpetcrusher** [comment]
Yeah, shouldn't be a challenge to get from receiving the error to identifying the resolution with the improved error messages. 

I put down the [ACME Basics](https://smallstep.com/docs/step-ca/acme-basics) page as one to have a note on in the interim, based off of my journey to find an internal ACME solution, as I went from Google -> Smallstep -> documentation/quick start -> docker install -> ACME basics.  I wasn't aware of the remote configuration concept since it was a "local" docker container and step-cli was happily working for simple certificates already.

Having this information in an issue alone should also help for now since it will be indexed by search engines.
